### PR TITLE
Refactor `langchain4j-vearch`

### LINKED ISSUE #1954: [FEATURE] Refactor `langchain4j-vearch`
**Is your feature request related to a problem? Please describe.**

The `langchain4j-vearch` currently uses old REST api to connect to `Vearch`, which has many limitations and is deprecated in new version of `Vearch`. Now `Vearch` has [new api](https://vearch.github.io/tools#/).

**Describe the solution you'd like**

Refactor `langchain4j-vearch` with new API.

**Describe alternatives you've considered**

No

**Additional context**

I will create a PR to refactor `langchain4j-vearch` in a few weeks. After refactoring, maybe the next step is to support filter in `langchain4j-vearch`.

### PR DESCRIPTION
**Pull request by Martin7-1**
## Issue

Closes #1954 

## Change

1. Refactor `langchain4j-vearch` with new [REST API](https://vearch.github.io/tools#/).
2. Support Metadata type: Long and Double.
3. Add request and response logging during http request.

> After this refactor, the next step is to support `filter` and `remove`.
>
> BTW, maybe this refactor is a breaking change because it change the constructor in `VearchEmbeddingStore`.

## General checklist

- [ ] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for changing existing embedding store integration

- [x] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 
 import org.testcontainers.containers.GenericContainer;
-import org.testcontainers.containers.wait.strategy.Wait;
 import org.testcontainers.utility.MountableFile;
 
 public class VearchContainer extends GenericContainer<VearchContainer> {
 
     public VearchContainer() {
-        super("vearch/vearch:3.4.1");
+        super("vearch/vearch:latest");
```
Comment (eddumelendez) [inline langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java:9]: I would use a fixed version instead. In the past it was hard to find the right version that was working with langchain4j.

Comment (Martin7-1) [reply langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java:9]: Good point. Will change it soon.

Comment (langchain4j) [reply langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java:9]: What about testing agains the latest version (to make sure we are up to date) and specify supported version(s) in the [documentation](https://docs.langchain4j.dev/integrations/embedding-stores/vearch)?

Comment (Martin7-1) [reply langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java:9]: Sounds great, I'm wondering if there is a way to start containers one by one and repeat the execution of the tests, thus ensuring that our tests are passing on multiple versions.

I'll try to implement it.

Comment (langchain4j) [reply langchain4j-vearch/src/test/java/dev/langchain4j/store/embedding/vearch/VearchContainer.java:9]: @katcharov already commented about it [here](https://discord.com/channels/1156626270772269217/1301238304536264818/1302014566770671678)

**Comment by langchain4j** [comment]
@Martin7-1 this might be a good candidate for the community repo as well. Moreover, you are the author of this module

**Comment by Martin7-1** [comment]
> @Martin7-1 this might be a good candidate for the community repo as well. Moreover, you are the author of this module

Sure. I think so.

**Comment by langchain4j** [comment]
@Martin7-1 great, thank you! Should I close this PR then?

**Comment by Martin7-1** [comment]
> @Martin7-1 great, thank you! Should I close this PR then?

Yes. Just go ahead.

**Pull request closed without merging**